### PR TITLE
DownloadManager: replace fini by destructor

### PR DESCRIPTION
**Pull request by HereThereBeDragons**
Issue #3095

### CODE REVIEW

**Comment by HereThereBeDragons** [comment]
@cernvm-bot cloudtest

**Comment by HereThereBeDragons** [comment]
@cernvm-bot cloudtest

File: cvmfs/network/download.h [hunk 1]
```diff
@@ -362,15 +360,15 @@ class DownloadManager {  // NOLINT(clang-analyzer-optin.performance.Padding)
    * request
    *
    * Sharding policy is shared between all download managers. As such shared
-   * pointers are used to allow for proper clean-up afterwards in Fini()
+   * pointers are used to allow for proper clean-up afterwards in destructor
```
Comment (jblomer) [inline cvmfs/network/download.h]:
```suggestion
   * pointers are used to allow for proper clean-up afterwards in the destructor
```

File: cvmfs/network/download.h [hunk 2]
```diff
@@ -362,15 +360,15 @@ class DownloadManager {  // NOLINT(clang-analyzer-optin.performance.Padding)
...
    * (We cannot assume the order in which the download managers are stopped)
    */
   SharedPtr<ShardingPolicy> sharding_policy_;
   /**
    * Health check for the proxies
    *
    * Health check is shared between all download managers. As such shared
-   * pointers are used to allow for proper clean-up afterwards in Fini()
+   * pointers are used to allow for proper clean-up afterwards in destructor
```
Comment (jblomer) [inline cvmfs/network/download.h]:
```suggestion
   * pointers are used to allow for proper clean-up afterwards in the destructor
```

File: cvmfs/network/download.cc [hunk 3]
```diff
@@ -1587,6 +1587,57 @@ bool DownloadManager::VerifyAndFinalize(const int curl_error, JobInfo *info) {
...
+  delete pool_handles_inuse_;
+  curl_multi_cleanup(curl_multi_);
+
+  delete header_lists_;
+  if (user_agent_)
+    free(user_agent_);
+
+  delete counters_;
+
+  if (opt_host_chain_) {
```
Comment (jblomer) [inline cvmfs/network/download.cc]: No `NULL` check around `delete` (also further down). It's already done by `delete`.

File: cvmfs/mountpoint.cc [hunk 4]
```diff
@@ -1914,11 +1914,9 @@ MountPoint::~MountPoint() {
   delete external_fetcher_;
   delete fetcher_;
   if (external_download_mgr_ != NULL) {
```
Comment (jblomer) [inline cvmfs/mountpoint.cc]: Remove `if`, too

File: cvmfs/mountpoint.cc [hunk 5]
```diff
@@ -1914,11 +1914,9 @@ MountPoint::~MountPoint() {
...
-    external_download_mgr_->Fini();
     delete external_download_mgr_;
   }
   if (download_mgr_ != NULL) {
```
Comment (jblomer) [inline cvmfs/mountpoint.cc]: Remove

**Review by jblomer — CHANGES_REQUESTED** [review]

**Comment by HereThereBeDragons** [comment]
@cernvm-bot cloudtest

**Review by jblomer — APPROVED** [review]

**Comment by vvolkl** [comment]
@cernvm-bot cloudtest

**Pull request merged**